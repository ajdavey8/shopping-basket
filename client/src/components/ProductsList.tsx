import React from 'react'

import { AddItem, RemoveItem, Product } from '../types';
import { List } from './styled-components'
import ProductsListItem from './ProductsListItem';

interface ProductsList {
  products: Product[]
  handleAddItem: (item: AddItem) => void
  handleRemoveItem: (item: RemoveItem) => void
}

const ProductsList = ({ products, handleAddItem, handleRemoveItem }: ProductsList) => (
  <List>
    {products.map((product: any) => (
      <ProductsListItem
        key={product.id}
        addItem={handleAddItem}
        removeItem={handleRemoveItem}
        {...product}
      />
    ))}
  </List>
)

export default ProductsList

