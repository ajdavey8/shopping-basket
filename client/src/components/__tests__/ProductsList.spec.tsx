import React from 'react';
import { render } from '@testing-library/react'
import ProductsList from '../ProductsList'

const addItem = jest.fn()
const removeItem = jest.fn()

const products = [
  {
    id: 1,
    name: 'soap',
    price: 2,
    image: 'pictureOfSoap',
    quantity: 1
  }
]

const setup = () => {
  const utils = render(
    <ProductsList
      products={products}
      handleAddItem={addItem}
      handleRemoveItem={removeItem}
    />
  )

  return {
    ...utils
  }
}

describe('<ProductsList>', () => {
  it('should return product', () => {
    const { asFragment } = setup()
    expect(asFragment).toMatchSnapshot();
  })
})