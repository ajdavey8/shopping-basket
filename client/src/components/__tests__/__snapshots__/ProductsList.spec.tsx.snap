// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductsList> should return product 1`] = `[Function]`;
